src/Core/Service/Subscriber/LicenseSyncSubscriber.php

```diff
@@ -103,32 +98,33 @@ public function serviceInstalled(AppInstalledEvent|AppUpdatedEvent $event): void
         }
 
         try {
-            $serviceEntry = $this->serviceRegistryClient->get($app->getName());
-            $this->syncLicenseByService($serviceEntry, $app, $context);
+            $this->syncLicenseByService($app, $context);
         } catch (\Throwable $e) {
             $this->logger->warning('Could not sync license', ['exception' => $e->getMessage()]);
         }
     }
 
-    private function syncLicenseByService(ServiceRegistryEntry $serviceEntry, AppEntity $app, Context $context, ?string $licenseKey = null, ?string $licenseHost = null): void
+    private function syncLicenseByService(AppEntity $app, Context $context, ?string $licenseKey = null, ?string $licenseHost = null): void
     {
-        if ($serviceEntry->licenseSyncEndPoint === null) {
-            return;
-        }
+        try {
+            $serviceEntry = $this->serviceRegistryClient->get($app->getName());
 
-        if ($licenseKey === null) {
-            $licenseKey = $this->config->getString(self::CONFIG_STORE_LICENSE_KEY);
-        }
+            if ($serviceEntry->licenseSyncEndPoint === null) {
+                return;
+            }
 
-        if ($licenseHost === null) {
-            $licenseHost = $this->config->getString(self::CONFIG_STORE_LICENSE_HOST);
-        }
+            if ($licenseKey === null) {
+                $licenseKey = $this->config->getString(self::CONFIG_STORE_LICENSE_KEY);
+            }
+
+            if ($licenseHost === null) {
+                $licenseHost = $this->config->getString(self::CONFIG_STORE_LICENSE_HOST);
+            }
```

Comment:
Can we fetch this in line 73/74 instead of doing it again for every app? 🤔

Reply:
Yeah, good point 🙈